test/e2e/es/auth_test.go

```diff
@@ -164,6 +166,30 @@ func TestESUserProvidedAuth(t *testing.T) {
 					return postDocument(b.Elasticsearch, k, esUser, writeIndexUpdated, expectedStatusCode)
 				}),
 			},
+			test.Step{
+				Name: "Update the secret to be a basic auth secret with a clear text password",
+				Test: test.Eventually(func() error {
+					var existingSecret corev1.Secret
+					if err := k.Client.Get(context.Background(), k8s.ExtractNamespacedName(&fileRealmSecret), &existingSecret); err != nil {
+						return err
+					}
+					existingSecret.Data = nil
+					existingSecret.StringData = map[string]string{
+						corev1.BasicAuthUsernameKey: sampleUser,
+						corev1.BasicAuthPasswordKey: samplePasswordCleartext,
+						filerealm.UsersRolesFile:    sampleUsersRolesFile,
+					}
+					return k.Client.Update(context.Background(), &existingSecret)
+				}),
+			},
+			test.Step{
+				Name: "ES API should eventually be accessible using the updated password and the updated role",
+				Test: test.Eventually(func() error {
+					esUser := esclient.BasicAuth{Name: sampleUser, Password: samplePasswordCleartext}
+					expectedStatusCode := 201
+					return postDocument(b.Elasticsearch, k, esUser, writeIndexUpdated, expectedStatusCode)
```

Comment:
🤦 Sorry, I didn't notice that the test actually failed:

```
=== RUN   TestESUserProvidedAuth/ES_API_should_eventually_be_accessible_using_the_updated_password_and_the_updated_role#01
Retries (5m0s timeout): ..........................
    step.go:43: 
        	Error Trace:	utils.go:88
        	Error:      	Received unexpected error:
        	            	unexpected status code 403
        	Test:       	TestESUserProvidedAuth/ES_API_should_eventually_be_accessible_using_the_updated_password_and_the_updated_role#01
```

Content of the error is:

```
action [indices:data/write/index] is unauthorized for user [myuser] with roles [], this action is granted by the index privileges [create_doc,create,index,write,all]
```

Reply:
My bad I should have run the e2e test on this PR. Thanks for catching this.